### DISCUSSION TITLE
Openzeppelin Files Not Found in chainlink-brownie-contracts

### DISCUSSION THREAD

**Discussion by vicjuma** [Cyfrin/foundry-full-course-cu#3044] [Q&A] [answered]
I am receiving a dependency for chainlink-brownie-contracts where some files for **openzeppelin** cannot be found. Here are the errors that are detected in Visual Studio Code _problems_ tab. The file that is affected is **KeeperRegistryBase1_3.sol**

```
Source "@openzeppelin/contracts/security/Pausable.sol" not found: File outside of allowed directories.
Source "@openzeppelin/contracts/security/ReentrancyGuard.sol" not found: File outside of allowed directories.
Source "@openzeppelin/contracts/utils/structs/EnumerableSet.sol" not found: File outside of allowed directories.
SPDX license identifier not provided in source file. Before publishing, consider adding a comment containing "SPDX-License-Identifier: <SPDX-License>" to each source file. Use "SPDX-License-Identifier: UNLICENSED" for non-open-source code. Please see https://spdx.org for more information.
```

I think this is why I am receiving an error ```server returned an error response: error code 3: execution reverted, data: "0x"``` on calling the FundMe getVersion function from Chainlink data feeds with ```cast call```. Here is my code

```
function getV() public view returns (uint256) {
        address chainlinkAddress = 0x694AA1769357215DE4FAC081bf1f309aDC325306;
        return AggregatorV3Interface(chainlinkAddress).version();
    }
```

I am using _chainlink-brownie-contracts version 1.3.0_, my cast version is 0.2.0. Kindly help. Thanks.

**Comment by heheboii11** [comment]
You need to install Openzeppelin lib, you cannot find ```ReentrancyGuard``` and all in the chainlink-brownie lib, 
please use this command in your terminal 
```forge install openzeppelin/openzeppelin-contracts --no-commit```

**Reply by vicjuma** [reply]
Thank you, the error is gone but now my cast call returns 0x. 

```
function getV() public view returns (uint256) {
        address chainlinkAddress = 0x694AA1769357215DE4FAC081bf1f309aDC325306;
        return AggregatorV3Interface(chainlinkAddress).version();
    }
```
![image](https://github.com/user-attachments/assets/6f14a17a-e5aa-43c4-8f1f-ff18f0248167)

**Reply by heheboii11** [reply]
Actually, function getV() function is not returning any value this is due to may be the contract did not deployed till now!! May be can you check whether did you deploy the contract or not ?

**Reply by EngrPips** [reply, marked as answer]
Hello @vicjuma, There is no chainlink price-feed contract on the Anvil Blockchain. I will advise that you follow Patrick as he teaches you, and later on, he will teach you how to fork a blockchain to run a test and how to test stuff locally by creating a Mock.

**Reply by vicjuma** [reply]
Hello @EngrPips , it is now that I understand that. Let me follow through. I got ahead of myself. Thanks muuch. Thanks too @heheboii11 . I am new to web3 bytheway and it is so amazing the milestones I have moved by following Patrick's tutorials and the Cyfrin community in general. Thank you all.

**Reply by EngrPips** [reply]
It's OK, Friend. Happens to the best of us, lol. Welcome aboard, and we are really glad to have you.

**Reply by Milles22** [reply]
I love the positive vibes on these question boards!
